wmi_check/datadog_checks/wmi_check/wmi_check.py

```diff
@@ -14,68 +15,49 @@ class WMICheck(WinWMICheck):
     Windows only.
     """
 
-    def __init__(self, name, init_config, agentConfig, instances):
-        WinWMICheck.__init__(self, name, init_config, agentConfig, instances)
-        self.wmi_samplers = {}
-        self.wmi_props = {}
-
-    def check(self, instance):
+    def __init__(self, name, init_config, instances):
+        # type: (str, Dict[str, Any], List[Dict[str, Any]]) -> None
+        super(WMICheck, self).__init__(name, init_config, instances)
+        self.custom_tags = self.instance.get('tags', [])  # type: List[str]
+        self.filters = self.instance.get('filters', [])  # type: List[Dict[str, WMIFilter]]
+        self.metrics_to_capture = self.instance.get('metrics', [])  # type: List[List[str]]
+        self.tag_by = self.instance.get('tag_by', "")  # type: str
+        self.tag_queries = self.instance.get('tag_queries', [])  # type: List[TagQuery]
+
+    def check(self, _):
+        # type: (Any) -> None
         """
         Fetch WMI metrics.
         """
-
-        # Connection information
-        host = instance.get('host', "localhost")
-        namespace = instance.get('namespace', "root\\cimv2")
-        provider = instance.get('provider')
-        username = instance.get('username', "")
-        password = instance.get('password', "")
-
-        # WMI instance
-        wmi_class = instance.get('class')
-        metrics = instance.get('metrics')
-        filters = instance.get('filters')
-        tag_by = instance.get('tag_by', "")
-        tag_queries = instance.get('tag_queries', [])
-
-        constant_tags = instance.get('constant_tags')
-        custom_tags = instance.get('tags', [])
-        if constant_tags is None:
-            constant_tags = list(custom_tags)
-        else:
-            constant_tags.extend(custom_tags)
+        constant_tags = self.instance.get('constant_tags', [])
+        if constant_tags:
             self.log.warning("`constant_tags` is being deprecated, please use `tags`")
+        constant_tags.extend(self.custom_tags)
 
         # Create or retrieve an existing WMISampler
-        instance_hash = hash_mutable(instance)
-        instance_key = self._get_instance_key(host, namespace, wmi_class, instance_hash)
-
-        metric_name_and_type_by_property, properties = self._get_wmi_properties(instance_key, metrics, tag_queries)
+        metric_name_and_type_by_property, properties = self.get_wmi_properties()
 
-        wmi_sampler = self._get_running_wmi_sampler(
-            instance_key,
-            wmi_class,
-            properties,
-            tag_by=tag_by,
-            filters=filters,
-            host=host,
-            namespace=namespace,
-            provider=provider,
-            username=username,
-            password=password,
-        )
+        wmi_sampler = self.get_running_wmi_sampler(properties, self.filters)
```

Comment:
I guess this should be done for every WMI integration right? 
Why was it changed?

Reply:
The new method was added here https://github.com/DataDog/integrations-core/pull/6329/files#diff-18a6bd2d960250c9f7f5d53bb88bdc37R253 although the old one is still in place. The reasoning is to have a cleaner interface. In agent 5 there were multiple check instances per check object, in agent 6 there is only one meaning that the object can hold all the config in memory instead of reliying on the passed in `instance`parameter, which we're trying to deprecate for agent 8.
There is a separate pr for win32_event_log but it may not be merged since @ofek is working on an alternate implementation.

Reply:
I see, thanks for the explanation 👍